torch/distributed/tensor/_sharding_prop.py

```diff
@@ -258,8 +259,12 @@ def propagate(self, op_info: OpInfo) -> None:
         # We cannot use an lru cache if we know that inputs will have dynamic shapes,
         # because SymInts are not hashable.
         # This is generally ok because this only happens during tracing in torch.compile,
-        # and tracing does not need to be as fast as eagermode DTensor usages.
-        if op_info.schema.has_symints:
+        # and and compile autograd initial tracing, whic do not need to be as fast as
+        # eagermode DTensor usages.
+        if (
+            _are_we_tracing()
+            or torch._dynamo.compiled_autograd.in_compiled_autograd_initial_trace
```

Comment:
would it be better if `are_we_tracing()` included compiled_autograd's flag in its logic, or is it actually important to keep these as separate flags users should check both of?